cmd/configure.ac

```diff
@@ -214,8 +214,8 @@ AC_ARG_WITH([host-arch-32bit-triplet],
 AC_SUBST(HOST_ARCH32_TRIPLET)
 AC_DEFINE_UNQUOTED([HOST_ARCH32_TRIPLET], "${HOST_ARCH32_TRIPLET}", [Arch triplet for 32bit libraries])
 
-SYSTEMD_SYSTEM_GENERATOR_DIR="$($PKG_CONFIG --variable=systemdsystemgeneratordir systemd)"
-AS_IF([test "x$SYSTEMD_SYSTEM_GENERATOR_DIR" = "x"], [SYSTEMD_SYSTEM_GENERATOR_DIR=/lib/systemd/system-generators])
+SYSTEMD_SYSTEM_GENERATOR_DIR="${prefix}$($PKG_CONFIG --variable=systemdsystemgeneratordir systemd)"
```

Comment:
I'm not super convinced this is something that needs to have a prefix. This is not our directory (hence not a local-snapd-installation type decision) but a systemd directory. Does systemd pick up generators from local directories?

Reply:
I found/usr/local/lib/systemd/system-generators/* in this documentation:

https://www.freedesktop.org/software/systemd/man/systemd.generator.html

And, if i run "sudo make uninstall": then my system generator was deleted :-(

Reply:
After discussing with Maciek I think he's right to point out that the prefix of snapd build and the system-wide install systemd directory path are unrelated.

If you want to get the "local" location it needs to be conveyed some other way, perhaps with a special case or in downstream packaging.

Reply:
I think the change is wrong. What happens is that the code runs pkg-config to determine which location the systemd package in the system expects to find the unit files and returns with an answer. There is no indication whether `systemd` will look for units files under `$prefix/$systemdsystemgeneratordir` (prefix can be a totally random thing) and AFAIK there's no way to ask whether such path is looked into by systemd.

If you just want to install all the files under some tree, pass `DESTDIR`, eg. `make DESTDIR=$PWD/install-dir install`.